java/gateway-core/src/main/java/com/avaulta/gateway/rules/JsonSchemaFilterUtils.java

```diff
@@ -185,10 +227,13 @@ private Object filterBySchema(String path, JsonNode provisionalOutput, JsonSchem
 
                         String key = entry.getKey();
                         JsonNode value = entry.getValue();
-                        JsonSchemaFilter propertySchema = schema.getProperties().get(key);
+                        JsonSchemaFilter propertySchema =
+                            schema.getProperties() == null ? null : schema.getProperties().get(key);
```

Comment:
that was the NPE, right?

Reply:
yeah, at least in the trace from yesterday